sysdeps/linux/include/bits/syscall.h

```diff
@@ -21,12 +21,14 @@ __sc_word_t __do_syscall7(long, __sc_word_t, __sc_word_t, __sc_word_t, __sc_word
 		__sc_word_t, __sc_word_t, __sc_word_t);
 long __do_syscall_ret(unsigned long);
 
+// We will need better syscall() macros in C++ because they cannot compiled with
+// libc++ 
+
+#warning C++ syscall() macros are disabled because of incompatibility with libc++
```

Comment:
This is a regression and I'd rather not merge that. Could you post the actual error when compiling ``libc++``?